index.js

```diff
@@ -309,52 +514,109 @@ Ycb.prototype = {
         var usedDimensions = {};
         var usedValues = {};
         var contexts = {};
+        var intervals = {};
         configLoop:
         for(var i=0; i<config.length; i++) {
-            if (config[i].settings) {
-                var setting = config[i].settings;
-                if(setting.length === 0 ) {
+            var configObj = config[i];
+            if(!configObj || configObj.constructor !== Object) {
+                logBundleWarning(i, 'non-object config', JSON.stringify(configObj));
+                continue;
+            }
+            if(!configObj.settings) {
+                if(!configObj.dimensions) {
+                    logBundleWarning(i, 'no valid settings field', JSON.stringify(configObj));
+                }
+                continue;
+            }
+            if(Array.isArray(configObj.settings)) { //convert old style settings to new style
+                configObj.settings = {dimensions: configObj.settings};
+            }
+            var setting = config[i].settings;
+
+            var settingDimensions = setting.dimensions;
+            var settingSchedule = setting.schedule;
+            var interval = undefined;
+            if(Object.keys(config[i]).length === 1) { //Don't skip as empty configs are valid and appear on tree walk
+                logBundleWarning(i, 'empty config', JSON.stringify(settingDimensions));
+            }
+            if(!Array.isArray(settingDimensions)) {
+                logBundleWarning(i, 'non-array settings', JSON.stringify(settingDimensions));
+                continue;
+            }
+            if(settingDimensions.length === 0 ) {
+                logBundleWarning(i, 'empty settings array');
```

Comment:
any more context we can add here so user can fix issue?

Reply:
It logs config index but unlike invalid dimension or similar there's no string to log that they can easily ctrl-f, unless we log the entire config object.